src/core/gosling.schema.ts

```diff
@@ -244,17 +244,33 @@ interface SingleTrackBase extends CommonTrackDef {
 
     // Mouse events
     experimental?: {
```

Comment:
Yes, I think it makes sense to turn these mouse events off by default. 

Since we currently have three mouse event listeners (i.e., `click`, `mouseover`, `rangeselect`), how about adding the following properties to enable using the events?

```js
experimental?: {
  // this will eventually be a track-level property, i.e., sibling of `mark`
  mouseEvent: boolean | {
    click: boolean,
    hover: boolean,
    rangeSelect: boolean,
    // or `enableClick`, `enableHover`, `enableRangeSelect`
    groupMarksByField?: string,
    ...
  }
}
```

This means that if users want to listen to mouse events through API, they will need to turn these on, e.g., `mouseEvent: { hover: true }` and `api.subscript('mouseover', ...)`.

Also, we can move the current styling objects to the `style` object:

```diff
experimental?: {
-  hovering: { color?: string, ...}
-  selection: { color?: string, ...}
-  brush: { color?: string, ...}
},
...,
style?: {
+  hover: { color?: string, ...}
+  select: { color?: string, ...}
+  brush: { color?: string, ...}
}
```

Reply:
(Btw, I gotta use a consistent term between hover and mouseover..)

Reply:
Since it's under the `experimental` key in the schema, I think we have some flexibility in deciding how to finally expose this behavior. Minor suggestion, but perhaps `mouseEvents` instead of `mouseEvent`. 

```typescript
type MouseEvent = 'click' | 'hover' | 'rangeSelect';
type MouseEvents = boolean | { [Event in MouseEvents]: boolean, groupMarksByField?: string };

{
  experimental?: {
    mouseEvents: MouseEvents;
  }
}
```

Reply:
And I agree, the key used under `mouseEvents` and the `api.subscribe(event)` should be identical

Reply:
Sound good. I will make this change in a follow-up PR.